Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/MotionEvents/EventDataEditor.cpp

```diff
@@ -36,23 +36,34 @@ namespace EMStudio
             return;
         }
 
-        const auto parent = pNode->GetParent();
-        if (parent && parent->GetSerializeContext()->CanDowncast(parent->GetClassMetadata()->m_typeId, azrtti_typeid<EMotionFX::EventData>(), parent->GetClassMetadata()->m_azRtti, nullptr))
+        auto parent = pNode->GetParent();
+        while (parent)
         {
-            const EMotionFX::EventData* eventData = static_cast<EMotionFX::EventData*>(parent->FirstInstance());
-            const AZ::Outcome<size_t> eventDataIndex = m_editor->FindEventDataIndex(eventData);
-            if (eventDataIndex.IsSuccess())
+            // Keep checking the parent node because we could be editing data in deeper layer.
+            // For example, eventData holding an array of elements, and we are editing the individual element.
+            if (parent->GetSerializeContext()->CanDowncast(
```

Comment:
question: Is there a test of anything closely related to this that's test to help catch and detect the problem that's being fixed?

Reply:
Okay fair enough, just was interested if we could test this added code to ensure it's doing what we expect, it's not as mission critical though so value isn't as high as some other areas, no worries